modules/tinymce/src/plugins/link/main/ts/ui/Controls.ts

```diff
@@ -95,7 +95,7 @@ const setupContextToolbars = (editor: Editor): void => {
       onSetup: Actions.toggleActiveState(editor)
     },
     label: 'Link',
-    predicate: (node) => !!Utils.getAnchorElement(editor, node) && Options.hasContextToolbar(editor),
+    predicate: (node) => Utils.isInAnchor(editor, node) && Options.hasContextToolbar(editor),
```

Comment:
I know this was like this before, but for performance this would be better if the order was swapped since getting the anchor by walking the DOM is much more over than checking if an option is enabled. Feel free to ignore this suggestion if you'd prefer since it's not related to this change.

```suggestion
    predicate: (node) => Options.hasContextToolbar(editor) && Utils.isInAnchor(editor, node),
```